### PR TITLE
CONSOLE-3654: console-plugin-template repository should support localization

### PR DESCRIPTION
**Pull request by Mylanos**
Links to: https://issues.redhat.com/browse/CONSOLE-3654

Adds possibility to generate locales in the console-plugin-template repo.

### CODE REVIEW

**Comment by jhadvig** [comment]
```
  1) Console plugin template test
       "before all" hook for "Verify the example page title":
     CypressError: `cy.exec('cd ../../console-plugin-template && /tmp/helm upgrade -i console-plugin-template charts/openshift-console-plugin -n console-plugin-template --create-namespace --set plugin.image=registry.build01.ci.openshift.org/ci-op-q9b5jsxw/pipeline@sha256:619b1c4b96723afa7f5e21994f8c8c2abf7ca8450b508a07aee2a9aeac392061')` timed out after waiting `60000ms`.
https://on.cypress.io/exec
Because this error occurred during a `before all` hook we are skipping the remaining tests in the current suite: `Console plugin template test`
Although you have test retries enabled, we do not retry tests when `before all` or `after all` hooks fail
```

/retest

**Comment by Mylanos** [comment]
{Operator unavailable (null): operator is not reporting conditions  Operator unavailable (null): operator is not reporting conditions}

{  openshift cluster install failed with cluster creation}

{  openshift cluster install failed overall}

File: README.md [hunk 1]
```diff
@@ -143,14 +143,50 @@ Additional parameters can be specified if desired. Consult the chart [values](ch
...
+Install the chart using the name of the plugin as the Helm release name into a new namespace or an existing namespace as specified by the `plugin_console-plugin-template` parameter and providing the location of the image within the `plugin.image` parameter by using the following command:
 
 ```shell
-helm upgrade -i  my-plugin charts/openshift-console-plugin -n my-plugin-namespace --create-namespace --set plugin.image=my-plugin-image-location
+helm upgrade -i  my-plugin charts/openshift-console-plugin -n plugin__console-plugin-template --create-namespace --set plugin.image=my-plugin-image-location
 ```
 
 NOTE: When deploying on OpenShift 4.10, it is recommended to add the parameter `--set plugin.securityContext.enabled=false` which will omit configurations related to Pod Security.
 
+NOTE: Please adhere to the specified namespace format `plugin__<name-of-the-plugin>` when defining namespaces. The name of the plugin should be extracted from the `consolePlugin` declaration within the [package.json](package.json) file.
```
Comment (jhadvig) [inline README.md:154]:
```suggestion
NOTE: When defining i18n namespace, adhere `plugin__<name-of-the-plugin>` format. The name of the plugin should be extracted from the `consolePlugin` declaration within the [package.json](package.json) file.
```

**Review by jhadvig — COMMENTED** [review]
One small nit, otherwise good to go 👍

**Review by jhadvig — APPROVED** [review]
/lgtm
/approve

**Comment by jhadvig** [comment]
QE Approver:
/assign @yanpzhan

Docs Approver:
/assign @opayne1

PX Approver:
/assign @RickJWagner

**Comment by yanpzhan** [comment]
I run the plugin using a local development environment by: 
In one terminal window, run:
yarn install
yarn run start
In another terminal window, run:
oc login 
yarn run start-console
Then access "localhost:9000/example?pseudolocalization=true&lng=en", could see contents are marked for i18n.
Screenshot from 2023-08-14 17-46-49
@Mylanos If there others I need to check for the US?

**Comment by Mylanos** [comment]
> I run the plugin using a local development environment by: In one terminal window, run: yarn install yarn run start In another terminal window, run: oc login yarn run start-console Then access "localhost:9000/example?pseudolocalization=true&lng=en", could see contents are marked for i18n. Screenshot from 2023-08-14 17-46-49 @Mylanos If there others I need to check for the US?

I'm sorry but I'm not sure if I understand correctly, could you please clarify the question?

**Comment by yanpzhan** [comment]
@Mylanos I mean except the check in above screenshot , if there are any other checkpoints I need to test for the feature from the pr code.

**Comment by Mylanos** [comment]
> @Mylanos I mean except the check in above screenshot , if there are any other checkpoints I need to test for the feature from the pr code.

That's pretty much it, other than that the features that this PR brings is the locale/translations files and contents get automatically generated from source codes via `yarn i18n`, so you can check if all the strings inside the source code get translated after you remove the locale files. But that is closely tied with the check if all the strings on the example page are marked for i18n.

**Pull request merged**